docs/source/cuda/memory.rst

```diff
@@ -156,6 +156,81 @@ traditional dynamic memory management.
 .. seealso::
    :ref:`Matrix multiplication example <cuda-matmul>`.
 
+Dynamic Shared Memory
+---------------------
+
+In order to use dynamic shared memory in kernel code declare shared array of size 0:
+
+.. code-block:: python
+
+   @cuda.jit
+   def kernel_func(x):
+      dyn_arr = cuda.shared.array(0, dtype=np.float32)
+      ...
+
+and specify the size of dynamic shared array in bytes during kernel invokation (dispatcher object instantiation):
+
+.. code-block:: python
+
+   kernel_func[32, 32, 0, 128](x)
+
+In above code dispatcher object is instantiated with 4 parameters:
+
+.. code-block:: python
+
+   kernel_func[grid_dim, block_dim, stream_id, dyn_shared_mem_size]
```

Comment:
The third parameter should be 0 or a stream object, rather than the ID of a stream, so I think it would be more illustrative to just refer to it as `stream`. (Perhaps in future we should encourage people to use `cuda.default_stream()` instead and warn when `0` is passed to mean the default stream, which is a historical idea and not very ergonomic IMHO)

```suggestion
   kernel_func[grid_dim, block_dim, stream, dyn_shared_mem_size]
```